pkg/controller/podautoscaler/replica_calculator.go

```diff
@@ -518,7 +517,17 @@ func calculatePodRequestsFromContainers(pod *v1.Pod, container string, resource
 			}
 			request += containerRequest.MilliValue()
 		}
+		// container names are unique inside the pod
+		if container == c.Name {
+			return request, nil
+		}
+	}
+
+	// If we're looking for a specific container and didn't find it
+	if container != "" {
```

Comment:
If you're planning to add this condition, I'll defer to @omerap12 for whether he wants that error condition or not, I'd expect a test case covering that error case. To ensure this change is properly maintained.